### PR TITLE
#9269 fix issue of correct map embed url in sharing map

### PR DESCRIPTION
**Pull request by mahmoudadel54**
## Description
In sharing map, if user select the Embed tab, the map embed link was pointing to "/#/viewer/{mapId}" instead of "/embedded.html#/{mapId}" and this was because of something wrong in the used regExp.
So I fixed the RegExp that corrects the embed map link. 

**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 
## Issue

**What is the current behavior?**
#9269 

**What is the new behavior?**
if user gos to the homepage or inside a map, clicks on the share button of the map card or from the side bar inside the map viewer, select the embed tab, user will get the correct embed url that points to /embedded.html#/{mapId}.
The embed url should point to the correct html page

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [x] No

### CODE REVIEW

**Review by allyoucanmap — APPROVED** [review]

File: web/client/components/share/SharePanel.jsx [hunk 1]
```diff
@@ -106,8 +106,8 @@ class SharePanel extends React.Component {
         modal: false,
         draggable: true,
         onClose: () => {},
-        shareUrlRegex: "(h[^#]*)#\\/viewer\\/([^\\/]*)\\/([A-Za-z0-9]*)",
-        shareUrlReplaceString: "$1embedded.html#/$3",
+        shareUrlRegex: "(h[^#]*)#\/viewer\/([^\/]*\/[A-Za-z0-9]*|[A-Za-z0-9]*)",
```
Comment (allyoucanmap) [inline web/client/components/share/SharePanel.jsx:109]: I approved but I have a doubt: are these changes really needed, in particular `\/[A-Za-z0-9]*|[A-Za-z0-9]*`, shouldn't be enough to remove `\\/([^\\/]*)` from the `shareUrlRegex` current on master?
Could you try with the follow one instead for shareUrlRegex prop?

`(h[^#]*)#\\/viewer\\/([A-Za-z0-9]*)`

**Review by allyoucanmap — CHANGES_REQUESTED** [review]

File: web/client/components/share/SharePanel.jsx (hunk 1 above)
Comment (mahmoudadel54) [reply web/client/components/share/SharePanel.jsx]: I was thinking the map URL may include 2 terms after viewer like: 
* http://hostname/#/viewer/shared/{map id}.
 
I have tried your suggestion and it works well

**Review by allyoucanmap — APPROVED** [review]
LGTM

**Comment by allyoucanmap** [comment]
@ElenaGallo please test this on dev and let us know if we can backport on 2023.02.xx, thanks

**Comment by ElenaGallo** [comment]
Test passed, @mahmoudadel54 please backport this PR to 2023.02.xx, thanks
